pandas/tests/test_window.py

```diff
@@ -2713,3 +2713,129 @@ def test_rolling_min_max_numeric_types(self):
             result = (DataFrame(np.arange(20, dtype=data_type))
                       .rolling(window=5).min())
             self.assertEqual(result.dtypes[0], np.dtype("f8"))
+
+
+class Grouping(tm.TestCase):
+
+    def setUp(self):
+        self.series = Series(np.arange(10))
+        self.frame = DataFrame({'A': [1] * 20 + [2] * 12 + [3] * 8,
+                                'B': np.arange(40)})
+
+    def test_mutated(self):
+
+        def f():
+            self.frame.groupby('A', foo=1)
+        self.assertRaises(TypeError, f)
+
+        g = self.frame.groupby('A')
+        self.assertFalse(g.mutated)
+        g = self.frame.groupby('A', mutated=True)
+        self.assertTrue(g.mutated)
+
+    def test_getitem(self):
+        g = self.frame.groupby('A')
+        g_mutated = self.frame.groupby('A', mutated=True)
+
+        expected = g_mutated.B.apply(lambda x: x.rolling(2).mean())
```

Comment:
so by our current rules this *wont'* return a multi-index (as its a regular groupby with a series returned, which effectively just get stacked). So for testing I *have* to return a multi-index which is the proper result. Note this is a ``Series`` groupby here (if you do the ``DataFrame`` it returns a multi-index.

This ``.apply`` DID not work in 0.18.0 (for a DataFrame), e.g.

This raised
``df.groupby('A').apply(lambda x: x.rolling(2).sum())``

This worked
``df.groupby('A').B.apply(lambda x: x.rolling(2).sum())``

So I kept it the same. 

cc @jorisvandenbossche